.github/workflows/operator-test.yaml

```diff
@@ -29,14 +29,30 @@ jobs:
       - name: Run chart-testing (install)
         run: ct install --charts charts/opentelemetry-operator
 
-      - name: Run KUTTL smoke tests
+      - name: Get opentelemetry-operator repository
+        run: |
+          appVersion=$(cat ./charts/opentelemetry-operator/Chart.yaml | sed -nr 's/appVersion: ([0-9]+\.[0-9]+\.[0-9]+)/\1/p')
+          git clone -b v"$appVersion" --single-branch https://github.com/open-telemetry/opentelemetry-operator.git ./opentelemetry-operator
+
+      - name: Use public target-allocator image in tests
+        run: |
+          LOCAL_TARGET_ALLOCATOR_IMG="local\/opentelemetry-operator-targetallocator:e2e"
+          PUBLIC_TARGET_ALLOCATOR_IMG="ghcr.io\/open-telemetry\/opentelemetry-operator\/target-allocator:0.1.0"
+
+          sed -i "s/$LOCAL_TARGET_ALLOCATOR_IMG/${PUBLIC_TARGET_ALLOCATOR_IMG}/g" ./opentelemetry-operator/tests/e2e/smoke-targetallocator/02-install.yaml
+          sed -i "s/$LOCAL_TARGET_ALLOCATOR_IMG/${PUBLIC_TARGET_ALLOCATOR_IMG}/g" ./opentelemetry-operator/tests/e2e/targetallocator-features/02-install.yaml
```

Comment:
This replaces all instances of the local target-allocator image with the public image right?  Would this be the charts permanent solution or is the image in the operator repo going to be switched to use the public version rendering this step obsolete some day?

Reply:
Yes, it replaces all local build images with public one. It will be permanent solution, in operator tests there will be always "development" image built during e2e tests.